packages/nextjs/pages/[contractAddress]/[network].tsx

```diff
@@ -57,26 +64,36 @@ const toCamelCase = (str: string) => {
 const ContractDetailPage = ({ addressFromUrl, chainIdFromUrl }: ServerSideProps) => {
   const router = useRouter();
   const { contractAddress, network } = router.query as ParsedQueryContractDetailsPage;
-  const [contractData, setContractData] = useState<ContractData>({ abi: [], address: contractAddress });
+  const [contractData, setContractData] = useState<ContractData>({ abi: [], address: contractAddress as Address });
+  const [localContractAbi, setLocalContractAbi] = useState<string>("");
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState<string | null>(null);
   const contractName = contractData.address;
-  const {
-    contractAbi: storedAbi,
-    setMainChainId,
-    chainId,
-    setImplementationAddress,
-  } = useAbiNinjaState(state => ({
-    contractAbi: state.contractAbi,
+  const { setMainChainId, chainId, setImplementationAddress } = useAbiNinjaState(state => ({
     setMainChainId: state.setMainChainId,
     chainId: state.mainChainId,
     setImplementationAddress: state.setImplementationAddress,
   }));
 
+  const { addChain, chains } = useGlobalState(state => ({
+    addChain: state.addChain,
+    chains: state.chains,
+  }));
+
   const publicClient = usePublicClient({
     chainId: parseInt(network),
   });
 
+  useEffect(() => {
+    const storedCustomChains = getStoredChainsFromLocalStorage();
+
+    storedCustomChains.forEach(chain => {
+      if (+network === chain.id) {
+        addChain(chain);
+      }
+    });
+  }, [addChain, network]);
```

Comment:
Maybe it makes more sense to set `chains` in global state as soon as app loads (In `_app.tsx`). Similar to how we are setting nativeCurrentPrice?